### PR TITLE
feat(login): add login components and graphql support

### PR DESCRIPTION
**Pull request by JeanBarriere**
- [x] add login modal (#196) 
- [x] add graphql dependencies (#205)
- [x] add avatar in navbar (#210)
- [x] add login process (#210)
- [x] e2e tests on login process (#270)

### CODE REVIEW

File: src/app/views/Invite.vue [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  }
+})
+export default class Invite extends Vue {
+  private email = ''
+  private emailError = ''
+
+  @Watch('email')
+  private onEmailHandler () {
+    if (
+      /^(([^<>()\\[\]\\.,;:\s@\\"]+(\.[^<>()\\[\]\\.,;:\s@\\"]+)*)|(\\".+\\"))@(([^<>()[\]\\.,;:\s@\\"]+\.)+[^<>()[\]\\.,;:\s@\\"]{2,})$/.test(
```
Comment (wilzbach) [inline src/app/views/Invite.vue]: Big no on this. No one will understand this in two weeks. That's the perfect reason to use a package like e.g. https://www.npmjs.com/package/email-validator

File: src/app/views/Login.vue [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  }
+})
+export default class Signup extends Vue {
+  private email = ''
+  private emailError = ''
+
+  @Watch('email')
+  private onEmailHandler () {
+    if (
+      /^(([^<>()\\[\]\\.,;:\s@\\"]+(\.[^<>()\\[\]\\.,;:\s@\\"]+)*)|(\\".+\\"))@(([^<>()[\]\\.,;:\s@\\"]+\.)+[^<>()[\]\\.,;:\s@\\"]{2,})$/.test(
```
Comment (wilzbach) [inline src/app/views/Login.vue]: And there's duplication (!)

File: src/internal/plugins/vue-apollo.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  const apolloProvider = new VueApollo({
+    defaultClient: apolloClient,
+    defaultOptions: {
+      $query: {
+        fetchPolicy: 'cache-and-network'
+      }
+    },
+    errorHandler (error: any) {
+      console.log('%cError', 'background: red; color: white; padding: 2px 4px; border-radius: 3px; font-weight: bold;', error.message)
```
Comment (wilzbach) [inline src/internal/plugins/vue-apollo.ts]: Why not simply `console.error`?

Comment (Arinono) [reply src/internal/plugins/vue-apollo.ts]: I don't have the exacte answer, but I guess we just want the error message and not the whole stack trace 🤷‍♂

File: src/app/views/Invite.vue (hunk 1 above)
Comment (Arinono) [reply src/app/views/Invite.vue]: I disagree a little bit with you on that. I can't read a RegExp like that, but in a method called `onEmailHandler` I'll always know what's this RegExp is doing

File: src/internal/plugins/vue-apollo.ts (hunk 3 above)
Comment (wilzbach) [reply src/internal/plugins/vue-apollo.ts]: `console.error(error.message)`

File: src/app/views/Invite.vue (hunk 1 above)
Comment (wilzbach) [reply src/app/views/Invite.vue]: >  I can't read a RegExp like that

Yup and there are almost no tests.
Also, this RegExp is very much incorrect.
Two quick examples:

image

image

File: src/internal/plugins/vue-apollo.ts (hunk 3 above)
Comment (Arinono) [reply src/internal/plugins/vue-apollo.ts]: Yes I know, what I was implying is that `console.error` is also giving the full stack trace, and `console.log` is not. I don't know if this is the reason, but this is the only difference I see.

Comment (wilzbach) [reply src/internal/plugins/vue-apollo.ts]: > console.error is also giving the full stack trace

That's optional and up to the IDE/browser on how to display. For example, by default, Chrome doesn't show the full stack trace, but let's you toggle it (another reason for `console.error`).

There's also the point that e.g. using custom CSS in log messages doesn't work well on the CLI nor with third-party logging tools like sentry.

File: src/app/views/Invite.vue (hunk 1 above)
Comment (JeanBarriere) [reply src/app/views/Invite.vue]: :+1: we should have some utils to merge these and prevent code duplication, but adding a dependency for a simple regex check is a bit overkill to me.

File: src/internal/plugins/vue-apollo.ts (hunk 3 above)
Comment (JeanBarriere) [reply src/internal/plugins/vue-apollo.ts]: This is generated by the vue cli plugin apollo and I honestly didn't checked:
https://github.com/Akryum/vue-cli-plugin-apollo/blob/master/generator/templates/vue-apollo/default/src/vue-apollo.js#L73

Comment (JeanBarriere) [reply src/internal/plugins/vue-apollo.ts]: The answer I guess is to provide a log and not an Error in the console, as graphql errors are to be interprated differently as a console.error message

Comment (JeanBarriere) [reply src/internal/plugins/vue-apollo.ts]: Screenshot from 2019-12-19 12-24-44
First one is a browser error, as Second one is just a log, removable, or to be dispatched on another layer (sentry or whatsoever)

File: src/app/views/Invite.vue (hunk 1 above)
Comment (wilzbach) [reply src/app/views/Invite.vue]: It's not simple as can be seen above. The first example that I tried was a false positive.

Comment (JeanBarriere) [reply src/app/views/Invite.vue]: I'm not saying we should keep the regex as it is, but in any case the `email-validator` package is just implementing another regex

**Pull request merged**